lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -617,10 +617,15 @@ else if(!request.keyName.equals(keyName))
 
 		/* timestamp */
 		if(request.timestamp == 0) {
-			if(options.queryTime)
-				request.timestamp = ably.time();
-			else
+			if (options.queryTime) {
+				/* Spec: RSA10k */
+				if (timeOffset == null) {
+					timeOffset = ably.time() - timestamp();
+				}
+				request.timestamp = timeOffset + timestamp();
+			} else {
 				request.timestamp = timestamp();
```

Comment:
This appears to be broken. You are not using the offset for subsequent requests so there is no point querying the time as it will not be used for any future requests when the local clock is out of sync